atomate2/openmm/jobs/core.py

```diff
@@ -0,0 +1,147 @@
+
+from typing import List, Union, Optional, Dict
+from pathlib import Path
+
+from dataclasses import dataclass
+
+from jobflow import Maker
+
+from pymatgen.io.openmm.schema import InputMoleculeSpec
+from pymatgen.io.openmm.sets import OpenMMSet
+
+from atomate2.openmm.jobs.base import BaseOpenmmMaker
+
+
+@dataclass
+class InputMaker(Maker):
+    name: str = "input maker"
+
+    def make(
+        self,
+        input_mol_dicts: List[Union[Dict, InputMoleculeSpec]],
+        density: Optional[float] = None,
+        box: Optional[List[float]] = None,
+        prev_dir: str | Path | None = None
+    ):
+        """
+        Create a job that generates an OpenMM input set.
```

Comment:
To confirm my understanding, `InputMaker` Job maker will be used to produce `OpenMMSet` objects from a set of directories?

Reply:
Exactly. This is a key difference from how the atomate2 vasp workflows work. 

In atomate2 for VASP, pretty much every job takes in a structure as input. For OpenMM, that doesn't work because a structure doesn't fully define the starting point of a simulation. You need the system, topology, integrator, and state to do that. The InputSet has all those things, so I think it makes sense to pass it around as the core input for most jobs (up for debate though).

However, actually creating an input set can take time. Calling `OpenMMSolutionGen.get_inputs(input_mol_dicts)`  to generate the `OpenMMSet` can take anywhere from a second to a minute to actually make the input set. If we are doing 1000 simulations, we don't want to do that locally. Instead we have the `InputMaker` job which delays the execution of `get_inputs` and makes it part of the workflow. We could imagine starting a flow with `InputMaker` or creating an `OpenMMSet` locally and feeding that directly into `NPTMaker`, or something like that.

Another core difference is that a VASP simulation is fully defined by a set of input files, e.g. a set of strings. Those define both the simulation input AND the simulation procedure. For OpenMM, the simulation input is defined by a set of input files but the simulation procedure is executed dynamically through the Python API. This means that if we want to keep track of the simulation procedure, we'll need some other way of recording what steps the simulation has followed.